01_fib/harirakr/main.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func fib(n int) []int {
+	if n >= 93 || n <= -93 {
+		fmt.Fprintln(out, "fib: Range out of bounds; takes input > -93 and < 93")
+		return nil
+	}
+	series := make([]int, 0)
+
+	sign := 1
+	if n < 0 {
+		sign = -1
+	}
+	a := 0
+	b := 0
+	for i := 0; i < n*sign; i++ {
+		sum := a + b*sign
+		if sum == 0 {
+			b = 1
+			series = append(series, a)
+			continue
+		}
+		a, b = b, sum
+		series = append(series, a)
```

Comment:
Initialise as `a, b := 0, 1`
Than you can collapse the loop body to
```go
a, b = b, a+sign*b
series = append(series, a)
```
This is aligned with lab requirements which (writely or not) asks you not to print the first `0` (see [lab assignment](https://github.com/anz-bank/go-course/tree/master/01_fib#lab-1---fibonacci))

Reply:
Done